File: Source/WickedSick.Fayde.Client.Engine/Typescript/Core/Style.ts
/// <reference path="DependencyObject.ts" />
/// CODE
/// <reference path="Setter.ts" />

module Fayde {
    export class Style extends XamlObject {
        private _IsSealed: bool = false;

        Setters: SetterCollection;
        BasedOn: Style;
        TargetType: Function;

        Seal() {
            if (this._IsSealed)
                return;
            this.Setters._Seal(this.TargetType);
            this._IsSealed = true;

            var base = this.BasedOn;
            if (base)
                base.Seal();
        }
    }
}

File: Source/Fayde.Server/Core/Style.cs
using Fayde.Primitives;
using Fayde.Xaml.Metadata;

namespace Fayde.Core
{
    [Element("Fayde")]
    public class Style : DependencyObject
    {
        public static readonly PropertyDescription SettersProperty = PropertyDescription.Register("Setters", typeof(DependencyObjectCollection<Setter>), typeof(Style), true);
        public static readonly PropertyDescription TargetTypeProperty = PropertyDescription.Register("TargetType", typeof(JsType), typeof(Style));
        public static readonly PropertyDescription BasedOnProperty = PropertyDescription.Register("BasedOn", typeof(Style), typeof(Style));
    }
}